apps/vmq_enhanced_auth/src/vmq_enhanced_auth.erl

```diff
@@ -283,29 +306,56 @@ check(Type, [Word | _] = Topic, User, SubscriberId) when is_binary(Word) ->
 
 check_all_acl(Type, TIn) ->
     {Tbl, _} = t(Type, all, TIn),
-    iterate_until_true(Tbl, fun(T) -> match(TIn, T) end).
+    iterate_until_true(Tbl, fun(T) ->
+        match_and_process_topic_metrics(TIn, T, Tbl, Type, T)
+    end).
 
 check_user_acl(Type, User, TIn) ->
     {Tbl, _} = t(Type, User, TIn),
     iterate_until_true(
-        ets:match(Tbl, {{User, '$1'}, '_'}),
-        fun([T]) -> match(TIn, T) end
+        ets:match(Tbl, {{User, '$1'}, '_', '_'}),
+        fun([T]) ->
+            Key = {User, T},
+            match_and_process_topic_metrics(TIn, T, Tbl, Type, Key)
+        end
     ).
 
 check_pattern_acl(Type, TIn, User, SubscriberId) ->
     {Tbl, _} = t(Type, pattern, TIn),
     iterate_until_true(Tbl, fun(P) ->
         T = topic(User, SubscriberId, P),
-        match(TIn, T)
+        match_and_process_topic_metrics(TIn, T, Tbl, Type, P)
     end).
 
 check_token_acl(Type, TIn, User, SubscriberId) ->
     {Tbl, _} = t(Type, token, TIn),
     iterate_until_true(Tbl, fun(P) ->
         T = topic(User, SubscriberId, P),
-        match(TIn, T)
+        match_and_process_topic_metrics(TIn, T, Tbl, Type, P)
```

Comment:
I have not seen any function name change due to addition of metric tracking. Can we let the function name as it was before?